core/src/View/Kokkos_ViewMapping.hpp

```diff
@@ -3313,25 +3313,32 @@ class ViewMapping<
   KOKKOS_INLINE_FUNCTION static void assign(
       ViewMapping<DstTraits, void>& dst,
       ViewMapping<SrcTraits, void> const& src, Args... args) {
-    static_assert(ViewMapping<DstTraits, traits_type, void>::is_assignable,
-                  "Subview destination type must be compatible with subview "
-                  "derived type");
```

Comment:
Please briefly explain why these compile-time assignability checks are removed.

Reply:
I removed because we are mapping to `traits_type` here, instead of to `DstTraits` like before, so the equiv of this check would be an identity map. Then below we call `ViewMapping<DstTraits, traits_type>::assign()`, which itself contains all the important checks for `traits_type` to `DstType`.